### PR TITLE
Add support for :maps_to parameter in mapping.json

### LINKED ISSUE #6: Content type name and field name must be equal when we have many_trough or many relation
Hello,

I have a mapping.json which looks like this:
```javascript
{
    "TxFdfactdbActivity": {
        "content_type": "User",
        "type": "entry",
        "fields": {
            "author_name": "name",
            "author_email": "email"
        },
        "links": {
        }
    },
    "ActivityLeg": {
        "content_type": "Game",
        "type": "entry",
        "fields": {
            "name": "name",
            "description": "description",
        },
        "links": {
            "many_through": [
                {
                    "relation_to": "TxFdfactdbActivity",
                    "primary_id": "activityId",
                    "foreign_id": "userId",
                    "through": "UserMap"
                }
            ]
        }
    }
}
```

and contentful_structure.json
```javascript
{
  "User": {
    "id": "3BedFYUJ3iUGoMyu4gEwIm",
    "name": "User",
    "description": null,
    "displayField": "name",
    "fields": {
      "name": "Text",
      "email": "Symbol"
    }
  },
  "Game": {
    "id": "1vssfcPAnyoCkyKK8swoWw",
    "name": "Game",
    "description": null,
    "displayField": "name",
    "fields": {
      "name": "Text",
      "Owners": {
        "id": "owners",
        "type": "Array",
        "link_type": "Entry",
        "link": "Link"
      }
    }
  }
}
```

When I run

`database-exporter --config-file settings.yml --prepare-json`

I see this error:
```
I, [2015-07-29T06:27:51.211160 #11416]  INFO -- : Mapping ActivityLeg - relation: many_through - [{"relation_to"=>"TxFdfactdbActivity", "primary_id"=>"activityId", "foreign_id"=>"userId", "through"=>"UserMap"}], record: 0
Missing link field for User in ActivityLeg in contentful structure JSON file!
 ```

It can be fixed if I rename the `Owners` field in the `Game` content type in contentful_structure.json to `User` (the same name as the `User` content type), or if I rename the `User` content type to `Owners`.

Could this be changed so the names of the content types and the names of the fields to be different when we have `many_through` and `manu` relations? I prefer not to change my content type name or my field name, because in some cases it makes more sense if they are different.

Thanks in advance.

### PR DESCRIPTION
**Pull request by dlitvakb**
Fixes #6

### CODE REVIEW

**Comment by neonichu** [comment]
Looks right to me, but because I'm not too familiar with the codebase, @pxlpnk should also have a look before merging.

File: lib/database/modules/relations_export.rb [hunk 1]
```diff
@@ -105,17 +105,17 @@ def model_content_type(model_name)
...
-        def contentful_field_attribute(model_name, associated_model, type)
+        def contentful_field_attribute(model_name, associated_model_attributes, type)
           contentful_model_hash(model_name)
           contentful_model_fields(model_name)
-          contentful_associated_model_name(model_name, associated_model)
-          contentful_associated_parameters(model_name, associated_model)
-          config.contentful_structure[model_content_type(model_name)][:fields][model_content_type(associated_model)][type]
+          contentful_associated_model_name(model_name, associated_model_attributes[:relation_to])
+          contentful_associated_parameters(model_name, associated_model_attributes)
+          config.contentful_structure[model_content_type(model_name)][:fields][get_associated_model(associated_model_attributes)][type]
```
Comment (pxlpnk) [inline lib/database/modules/relations_export.rb]: there is a log going on in this line. That could be a bit simplified

File: lib/database/modules/relations_export.rb [hunk 2]
```diff
@@ -130,8 +130,9 @@ def contentful_associated_model_name(model_name, associated_model)
           fail ArgumentError, "Missing associated model content type name for #{model_name} in MAPPING JSON file" unless model_content_type(associated_model)
         end
 
-        def contentful_associated_parameters(model_name, associated_model)
-          fail ArgumentError, "Missing link field for #{model_content_type(associated_model)} in #{model_name} in contentful structure JSON file!" unless config.contentful_structure[model_content_type(model_name)][:fields][model_content_type(associated_model)]
+        def contentful_associated_parameters(model_name, associated_model_attributes)
+          associated_model = get_associated_model(associated_model_attributes)
+          fail ArgumentError, "Missing link field for #{model_content_type(associated_model)} in #{model_name} in contentful structure JSON file!" unless config.contentful_structure[model_content_type(model_name)][:fields][associated_model]
```
Comment (pxlpnk) [inline lib/database/modules/relations_export.rb]: This condition also could be made a bit clearer once we already touch it.

File: spec/lib/database/relations_export_spec.rb [hunk 3]
```diff
@@ -85,15 +85,28 @@ module Database
...
           it 'id' do
-            result = @exporter.contentful_field_attribute('Profiles', 'Users', 'id')
+            result = @exporter.contentful_field_attribute('Profiles', {relation_to: "Users", foreign_id: "user_id"}, 'id')
             expect(result).to eq 'user'
           end
         end
 
+        context 'get_associated_model' do
+          it 'calls get_associated_model' do
+            expect_any_instance_of(Export).to receive(:get_associated_model).with({relation_to: "Users", foreign_id: "user_id"}) { "User" }.twice
```
Comment (pxlpnk) [inline spec/lib/database/relations_export_spec.rb]: Calling the same thing twice seems a bit like a code smell here

File: spec/lib/database/relations_export_spec.rb [hunk 4]
```diff
@@ -144,7 +157,7 @@ module Database
         end
 
         it 'map_many_association' do
-          expect_any_instance_of(Export).to receive(:contentful_field_attribute).with('model_name', 'Comments', :id) { 'ct_field_id' }
+          expect_any_instance_of(Export).to receive(:contentful_field_attribute).with('model_name', {:relation_to=>"Comments", :primary_id=>"job_add_id"}, :id) { 'ct_field_id' }
```
Comment (pxlpnk) [inline spec/lib/database/relations_export_spec.rb]: Prefer new hash syntax.

File: spec/lib/database/relations_export_spec.rb [hunk 5]
```diff
@@ -165,7 +178,7 @@ module Database
                                                 {'entry' => 'value'},
                                                 'entry_path',
                                                 'related_to')
-          expect_any_instance_of(Export).to receive(:contentful_field_attribute).with('model_name', 'Comments', :id) { 'ct_field_id' }
+          expect_any_instance_of(Export).to receive(:contentful_field_attribute).with('model_name', {:relation_to=>"Comments", :primary_id=>"job_add_id"}, :id) { 'ct_field_id' }
```
Comment (pxlpnk) [inline spec/lib/database/relations_export_spec.rb]: Prefer new hash syntax.

**Pull request merged**